game-core/src/main/java/games/strategy/triplea/ui/MacOsIntegration.java

```diff
@@ -39,34 +39,63 @@ public static void addAboutHandler(final Runnable handler) {
+  }
+
+  private static boolean isJavaVersionAtLeast9() {
+    return isJavaVersionAtLeast9(SystemProperties.getJavaSpecificationVersion());
+  }
+
+  @VisibleForTesting
+  static boolean isJavaVersionAtLeast9(final String encodedJavaSpecificationVersion) {
+    try {
+      return Float.parseFloat(encodedJavaSpecificationVersion) >= 9.0F;
+    } catch (final NumberFormatException e) {
+      log.log(Level.SEVERE, "Malformed Java specification version: '" + encodedJavaSpecificationVersion + "'", e);
+      return false;
+    }
   }
 
   private static void addHandler(
       final String handlerClassName,
       final String handlerMethodName,
       final String addHandlerMethodName,
-      final ThrowingConsumer<Object[], Throwable> handler) {
+      final ThrowingConsumer<Object[], Exception> handler) {
     try {
-      final Class<?> applicationClass = Class.forName("com.apple.eawt.Application");
-      final Object application = applicationClass.getConstructor((Class[]) null).newInstance((Object[]) null);
-      final Class<?> handlerClass = Class.forName(handlerClassName);
-      final Object handlerProxy = Proxy.newProxyInstance(
-          MacOsIntegration.class.getClassLoader(),
-          new Class<?>[] {handlerClass},
-          (proxy, method, args) -> {
-            if (handlerMethodName.equals(method.getName())) {
-              handler.accept(args);
-            }
-            return null;
-          });
-      applicationClass.getDeclaredMethod(addHandlerMethodName, handlerClass).invoke(application, handlerProxy);
+      addHandler(newApplication(), handlerClassName, handlerMethodName, addHandlerMethodName, handler);
     } catch (final ReflectiveOperationException e) {
-      log.log(Level.SEVERE, "Failed to add macOS application handler (" + handlerClassName + ")", e);
+      log.log(Level.SEVERE, "Failed to add macOS application handler: " + handlerClassName, e);
     }
   }
 
+  @VisibleForTesting
+  static void addHandler(
+      final Object application,
+      final String handlerClassName,
+      final String handlerMethodName,
+      final String addHandlerMethodName,
+      final ThrowingConsumer<Object[], Exception> handler) throws ReflectiveOperationException {
+    final Class<?> handlerClass = Class.forName(handlerClassName);
+    final Object handlerProxy = Proxy.newProxyInstance(
+        MacOsIntegration.class.getClassLoader(),
+        new Class<?>[] {handlerClass},
+        (proxy, method, args) -> {
+          if (handlerMethodName.equals(method.getName())) {
+            handler.accept(args);
+          }
+          return null;
+        });
+    application.getClass()
+        .getDeclaredMethod(addHandlerMethodName, handlerClass)
+        .invoke(application, handlerProxy);
+  }
+
+  private static Object newApplication() throws ReflectiveOperationException {
+    return Class.forName("com.apple.eawt.Application")
+        .getConstructor((Class[]) null)
+        .newInstance((Object[]) null);
```

Comment:
Is there a reason null is being passed instead of passing 0 parameters?

Reply:
I just copy-pasted it from the `keystore-explorer` project. :smile:  I should be able to remove the arguments.

Reply:
I suspect the `keystore-explorer` code is passing `null` in an attempted optimization to avoid allocation of a zero-length array.

Reply:
60fa17305